lc-hax/Scripts/Modules/AntiKickMod.cs

```diff
@@ -36,10 +41,24 @@ public sealed class AntiKickMod : MonoBehaviour {
 
         this.HasAnnouncedGameJoin = true;
 
-        Helper.CreateComponent<WaitForBehaviour>().SetPredicate(time => time > 1.0f).Init(() => {
-            Chat.Announce($"{Helper.LocalPlayer?.playerUsername} disconnected.", true);
-            Chat.Print("You are invisible! Do /invis to disable!");
-        });
+        if (!this.ClearChatExecuted) {
+            Helper.CreateComponent<WaitForBehaviour>().SetPredicate(time => time > 1.0f).Init(() => {
+                Chat.Announce($"{Helper.LocalPlayer?.playerUsername} disconnected.", true);
+                Chat.Print("You are invisible! Do /invis to disable!");
+            });
+        }
+
+        if (this.ClearChatExecuted) {
+            this.ClearChatExecuted = false;
+            Helper.CreateComponent<WaitForBehaviour>().SetPredicate(time => time > 1.0f).Init(Chat.Clear);
+        }
+    }
+
+    IEnumerator DelayClearChatToFalse() {
+        yield return new WaitForSeconds(6.5f);
+        if (Helper.LocalPlayer is null) {
+            this.ClearChatExecuted = false;
+        }
```

Comment:
What's this for?

Reply:
If it fails to join anyways, then isn’t it alright even if it clears the chat?

Reply:
> If it fails to join anyways, then isn’t it alright even if it clears the chat?

I guess so, the idea is when you first join it looks like you disconnected instantly, but if you get kicked it clears to join message from the chat to prevent suspicion.

Reply:
I don’t think that would work. If you get kicked, you lose connection to the server and the RPC to clear the chat will no longer work.